ic-http/src/lib.rs

```diff
@@ -125,42 +125,67 @@ pub mod mock;
 pub use crate::http_request::http_request;
 pub use crate::request::create_request;
 pub use crate::response::create_response;
-pub use crate::transform::TransformFn;
 
-use crate::mock::{hash, Mock};
-use ic_cdk::api::management_canister::http_request::CanisterHttpRequestArgument;
-use std::collections::HashMap;
-use std::sync::RwLock;
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
Almost everything in this file is under the `#[cfg(not(target_arch = "wasm32"))]` directive. Consider moving that entire logic into a separate module so that we don't need to specify this directive for every line.

Reply:
Done, thank you, Islam!